recipes/squash/build.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -ex
+
+EXTRA_CMAKE_ARGS=""
+if [[ `uname` == 'Darwin' ]]; then
+  export LIBRARY_SEARCH_VAR=DYLD_FALLBACK_LIBRARY_PATH
+  # We override most toolchain variables to modify the min version for OSX.
+  export MACOSX_VERSION_MIN="10.7"
```

Comment:
[toolchain](https://github.com/conda-forge/toolchain-feedstock/blob/master/recipe/activate.sh) uses min version `10.9`. Then `-stdlib=libc++`, `-headerpad_max_install_names' and `-lc++` are used in `toolchain` too.

I have reviewed other recipes and references like:
https://github.com/conda-forge/vigra-feedstock/blob/master/recipe/build.sh
https://github.com/conda-forge/protobuf-feedstock/blob/master/recipe/build.sh
https://github.com/conda-forge/flatbuffers-feedstock/blob/master/recipe/build.sh
https://ipscc.readthedocs.io/en/latest/build-sh-guidelines.html#clang-and-libc-dylib

And it makes sense to me to target older OSX version. Unless official conda-forge policy is to go with `toolchain` defaults, if that so I will remove this change.